hooks/shapediver/viewer/interaction/useProcessPattern.ts

```diff
@@ -0,0 +1,37 @@
+import { useShapeDiverStoreViewer } from "shared/store/useShapeDiverStoreViewer";
+import { processPattern } from "./utils/patternUtils";
+import { useState, useEffect } from "react";
+
+// #region Functions (1)
+
+/**
+ * Hook that processes a pattern for a session.
+ * 
+ * @param sessionId The ID of the session.
+ * @param nameFilter The name filter to apply to the pattern.
+ */
+export function useProcessPattern(sessionId: string, nameFilter?: string[]): {
```

Comment:
@MajorMeerkatThe3rd I suggest renaming this to sth like `useCreateNameFilterPattern`